### ISSUE TITLE
Support Auto Light/Dark Mode on macOS

### ISSUE THREAD

**Issue by jzelinskie** [mumble-voip/mumble#4420]
**Context**
macOS users with the "Auto" Appearance configuration set (which is the default) have all of their applications transition between Light and Dark modes at sunset/sunrise.

**Describe the feature you have in mind**
Mumble already supports a Light and Dark theme that fit with the OS themes well, it need only switch between them at the right time.

**Describe alternatives you've considered**
This might be something for qt to handle, rather than Mumble itself.

**Additional context**
* [Apple Developer Doc](https://developer.apple.com/documentation/xcode/supporting_dark_mode_in_your_interface)

**Comment by Krzmbrzl** [comment]
From what I find on the web it seems like this kind of feature is supported from Qt 5.12 onwards. I think however that our explicit theming breaks the automatic transitions and I suppose that is why it's not showing.

Switching themes while Mumble is running is not possible either as that requires it to be restarted so I'm not sure if that'd make for a great experience...

What version of Mumble were you testing? And what Qt version was it built with (Can be found out in the Help dropdown menu)?

**Comment by jzelinskie** [comment]
>I think however that our explicit theming breaks the automatic transitions and I suppose that is why it's not showing.
>Switching themes while Mumble is running is not possible either as that requires it to be restarted so I'm not sure if that'd make for a great experience...

Agreed. However, I'd reckon that majority of macOS users would probably prefer to forgo explicit theming in order to support this functionality. Having a checkbox enabling theming or setting the theme in the dropdown to "None" could be used as the flag to fallback to this qt functionality.

>What version of Mumble were you testing? And what Qt version was it built with (Can be found out in the Help dropdown menu)?

I wasn't using anything crazy -- just latest stable release, 1.3.2, which ships Qt 5.6.2. I don't mind building something to test behavior if the development team does not have any macs for testing.

**Comment by Krzmbrzl** [comment]
> I wasn't using anything crazy -- just latest stable release, 1.3.2, which ships Qt 5.6.2. I don't mind building something to test behavior if the development team does not have any macs for testing.

That'd be great as my findings were that this feature got introduced with Qt 5.12. So _maybe_ it could be working (a bit?) after all ^^
I think it shouldn't matter whether you build the current master branch or the 1.3.x branch as nothing changed in regards to theming.

The build instructions are located at https://wiki.mumble.info/wiki/BuildingMacOSX but looking at them I think they are outdated. Instead you might want to use the build scripts we use on our macOS CI (these only work with the master branch though as they're using cmake. If you want to build 1.3.x, then I can point you to the old build scripts in that branch):
- [Installing the build env](https://github.com/mumble-voip/mumble/blob/master/.ci/azure-pipelines/install-environment_macos.bash)
- [Compiling](https://github.com/mumble-voip/mumble/blob/master/.ci/azure-pipelines/build_macos.bash)

**Comment by davidebeatrici** [comment]
Maybe we could add a flag to each theme that indicates whether it's dark/light/neutral and then decide which one to set depending on the system's.

**Comment by rakleed** [comment]
It would be great if you could add support for this functionality to your Windows app.

**Comment by TerryGeng** [comment]
> Maybe we could add a flag to each theme that indicates whether it's dark/light/neutral and then decide which one to set depending on the system's.

For macOS: https://successfulsoftware.net/2021/03/31/how-to-add-a-dark-theme-to-your-qt-application/

**Comment by Hartmnt** [comment]
Implemented in #6877 
cheers!
